src/operator/every.ts

```diff
@@ -18,22 +16,21 @@ export function every<T>(predicate: (value: T, index: number, source: Observable
                          thisArg?: any): Observable<boolean> {
   const source = this;
   if (source._isScalar) {
```

Comment:
We need to remove all eager predicate calls in here. This will hurt perf for Array and Scalar types slightly, but will improve predictability. The fact it's still in here is an oversight.